v2/outbound_transport_middleware.go

```diff
@@ -72,17 +72,28 @@ func applyUnaryOutboundTransportMiddleware(o UnaryOutbound, f UnaryOutboundTrans
 	return unaryOutboundWithMiddleware{o: o, f: f}
 }
 
-// UnaryOutboundTransportMiddlewareFunc adapts a function into a UnaryOutboundTransportMiddleware.
-type UnaryOutboundTransportMiddlewareFunc func(context.Context, *Request, *Buffer, UnaryOutbound) (*Response, *Buffer, error)
+// UnaryOutboundTransportMiddlewareStruct adapts a function and name into a
+// UnaryOutboundTransportMiddleware.
+type UnaryOutboundTransportMiddlewareStruct struct {
```

Comment:
Rather than naming this `*Struct`, what are your thoughts on keeping the naming convention and adding a new constructor for `UnaryOutboundTransportMiddlewareFunc`? 

```
type UnaryOutboundTransportMiddlewareFunc struct {
  Name string
  Func func(context.Context, *Request, *Buffer, UnaryOutbound) (*Response, *Buffer, error)
}

func NewUnaryOutboundTransportMiddlewareFunc(
  name string,
  fn func(context.Context, *Request, *Buffer, UnaryOutbound) (*Response, *Buffer, error),
) UnaryOutboundTransportMiddlewareFunc { ... }
```

Reply:
That seems like a misnomer for the struct.

Let's make the struct private and drop the Func. We can expose this as a constructor convenience